src/http.cpp

```diff
@@ -684,21 +665,20 @@ m_off_t SpeedController::calculateSpeed(long long numBytes)
 
     while (transferBytes.size())
```

Comment:
I'd replace this loop by a different version, which is probably more efficient since it doesn't create a new iterator for each iteration of the loop:
```
    map<dstime, m_off_t>::iterator it = transferBytes.begin();
    while (it != transferBytes.end())
    {
        dstime dt = it->first;
        if (currentTime - dt < SPEED_MEAN_INTERVAL_DS)
        {
            break;
        }

        partialBytes -= it->second;
        it++;
        transferBytes.erase(dt);
    }
```